SRR.UIP.HW9.LandCalculator.BLL/Services/PointsValidator.cs

```diff
@@ -0,0 +1,143 @@
+﻿using SRR.UIP.HW9.LandCalculator.Shared.Interfaces;
+using SRR.UIP.HW9.LandCalculator.Shared.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace SRR.UIP.HW9.LandCalculator.BLL.Services
+{
+    public class PointsValidator : IPointsValidator
+    {
+        public PointsValidationResult GetValidationResult(List<Point> points)
+        {
+            PointsValidationResult validationResult = new PointsValidationResult();
+            validationResult.Mistakes = PointsMistakes.Nothing;
+            if (!IsPointsCountMoreThanTwo(points))
+            {
+                validationResult.Mistakes = PointsMistakes.NotEnoughUniquePoints;
+                validationResult.ArePointsValid = false;
+                validationResult.Message = $"Passed validation {validationResult.ArePointsValid}; Problems: {validationResult.Mistakes}";
+                return validationResult;
+            }
+            bool AreDublicates = GetNumberOfDublicates(points) != 0;
+            validationResult.Mistakes = AreDublicates ?
+                validationResult.Mistakes | PointsMistakes.NotEnoughUniquePoints : validationResult.Mistakes;
+            validationResult.Mistakes = IsClosedContour(points) ?
```

Comment:
why did you use separate variable for `GetNumberOfDublicates(points) != 0` but didn't do this IsClosedContour(points) ? please align your approach